Packs/GoogleCloudLogging/Integrations/GoogleCloudLogging/GoogleCloudLogging.py

```diff
@@ -0,0 +1,285 @@
+        :param insecure:
+        """
+        try:
+            credentials = service_account.Credentials.from_service_account_info(info=client_secret, scopes=SCOPES)
+            if proxy or insecure:
+                http_client = google_auth_httplib2.AuthorizedHttp(
+                    credentials, http=self.get_http_client_with_proxy(proxy, insecure))
+                self.service = discovery.build('logging', 'v2', http=http_client)
+            else:
+                self.service = discovery.build('logging', 'v2', credentials=credentials)
+        except Exception as e:
+            raise e
+
+    # disable-secrets-detection-start
+
+    @staticmethod
+    def get_http_client_with_proxy(proxy, insecure):
+        """
+        Create an http client with proxy with whom to use when using a proxy.
+        :param proxy: Whether to use a proxy.
+        :param insecure: Whether to disable ssl and use an insecure connection.
+        :return:
+        """
+        if proxy:
+            proxies = handle_proxy()
+            https_proxy = proxies.get('https')
+            http_proxy = proxies.get('http')
+            proxy_conf = https_proxy if https_proxy else http_proxy
+            # if no proxy_conf - ignore proxy
+            if proxy_conf:
+                if not proxy_conf.startswith('https') and not proxy_conf.startswith('http'):
+                    proxy_conf = 'https://' + proxy_conf
+                parsed_proxy = urllib.parse.urlparse(proxy_conf)
+                proxy_info = httplib2.ProxyInfo(
+                    proxy_type=httplib2.socks.PROXY_TYPE_HTTP,
+                    proxy_host=parsed_proxy.hostname,
+                    proxy_port=parsed_proxy.port,
+                    proxy_user=parsed_proxy.username,
+                    proxy_pass=parsed_proxy.password)
+                return httplib2.Http(proxy_info=proxy_info, disable_ssl_certificate_validation=insecure)
+        return httplib2.Http(disable_ssl_certificate_validation=insecure)
+
+    def get_entries_request(self, request_body: dict) -> dict:
+        """
+        Gets a request body and execute the request.
+
+        Args:
+            request_body (dict): The request body.
+
+        Returns:
+            The request response.
+        """
+        return self.service.entries().list(body=request_body).execute()  # pylint: disable=E1101
+
+        # disable-secrets-detection-end
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: GoogleCloudLoggingClient) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        message = 'ok'
+    except Exception as e:
```

Comment:
It will never reach here. Why do we need this? 
Create a function called `prepare_gsuite_client` and check it instead. (find example in GsuiteAuditor)